.github/workflows/ci.yaml

```diff
@@ -0,0 +1,101 @@
+name: Publish to ECR
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - "helm/**"
+      - "**.md"
+      - "**.txt"
+      - "Jenkinsfile"
+      - ".git**"
+
+  pull_request:
+    paths-ignore:
+      - "helm/**"
+      - "**.md"
+      - "**.txt"
+      - "Jenkinsfile"
+      - ".git**"
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    env:
+      RAILS_ENV: test
+      GOVUK_CONTENT_SCHEMAS_PATH: vendor/govuk-content-schemas
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Checkout GOV.UK Content Schemas
+        uses: actions/checkout@v2
+        with:
+          repository: alphagov/govuk-content-schemas
+          ref: deployed-to-production
+          path: vendor/govuk-content-schemas
+
+      - name: Setup Ruby
+        uses: ruby/setup-ruby@v1
+
+      - name: Retrieve cached Gems
+        uses: actions/cache@v2
+        with:
+          path: vendor/bundle
+          key: bundle-v2-${{ hashFiles('**/Gemfile.lock') }}
+          restore-keys: bundle-v2
+
+      - name: Install Ruby dependencies
+        run: bundle install --jobs 4 --retry 3 --deployment
+
+      - name: Setup Node
+        uses: actions/setup-node@v1
+
+      - name: Retrieve cached node modules
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: yarn
+
+      - name: Install JavaScript dependencies
+        run: yarn install --frozen-lockfile
+
+      - name: Precompile assets
+        run: bundle exec rails assets:precompile
+
+      - name: Run tests
+        run: bundle exec rake
+
+  publish:
+    needs: test
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ECR_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_ECR_SECRET_ACCESS_KEY }}
```

Comment:
Since these are org level (GDS) secrets do we need to prefix them to indicate that they are GOVUK specific? I assume there's other programmes in GDS that may also want an ECR secret and it'd be good if it's clear this one belongs to GOV.UK (that's me assuming that we're using separate ECR repos)